src/languages/typescript.ts

```diff
@@ -61,6 +61,16 @@ function collapseWhitespace(source: string, range: Range): Range {
     };
 }
 
+function compact<T>(items: (T | undefined)[]): T[] {
+    const compactedItems: T[] = [];
+    for (const item of items) {
+        if (item !== undefined) {
+            compactedItems.push(item);
+        }
+    }
+    return compactedItems;
+}
```

Comment:
Not sure if there is a better type-safe way to achieve `(T | undefined)[]` -> `T[]`. `.filter(i => i !== undefined)` doesn't do it and didn't want to `as T[]`

Reply:
I've used `.filter(i => i!)`. Seems okay as long as there is a clear step above which makes it obviously defined.